src/app/icd/ICDManager.cpp

```diff
@@ -228,5 +254,74 @@ void ICDManager::OnTransitionToIdle(System::Layer * aLayer, void * appState)
     pIcdManager->mStateObserver->OnTransitionToIdle();
 }
 
+bool ICDManager::EnsureChipStackLock()
```

Comment:
please make this some form of RAII object - tracing that lock and unlock are paired is hard otherwise.

I would also generally prefer us NOT needing to do this: it is very convenient to just say "lock if needed" however that convenence means that code paths will not be as carefully reviewed and we would not know if a lock is needed or not. Also note that darwin does NOT support stack locking at all. even if we do not expect ICD production devices to run on mac, test programs should be runnable.

@bzbarsky-apple for apple opinion

Reply:
remove this internal lock mechanism from the ICDManager. the ICDNotifier is expected to be used from the chipT ask context